test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/HistoryTests.cs

```diff
@@ -194,52 +194,78 @@ public async Task GivenAValueForSinceAndBeforeWithModifications_WhenGettingSyste
         [Fact]
         public async Task GivenAValueForSinceAndBeforeCloseToLastModifiedTime_WhenGettingSystemHistory_TheServerShouldNotMissRecords()
         {
-            var tag = Guid.NewGuid().ToString();
+            // as this test is timing dependent, we will try it three times
+            // to see if it can pass
+            for (int i = 0; i < 3; i++)
+            {
+                bool success = false;
+                var newResources = new List<Resource>();
 
-            var since = await GetStartTimeForHistoryTest(tag);
+                try
+                {
+                    var tag = Guid.NewGuid().ToString();
 
-            var newResources = new List<Resource>();
+                    var since = await GetStartTimeForHistoryTest(tag);
```

Comment:
>GetStartTimeForHistoryTest [](http://example.com/codeflow?start=38&length=26)

can you do this instead?
(not in this line but after all resource created)
newResources.Select(x=>x.Meta.LastUpdated.Value).Min()